packages/pluggableWidgets/sidebar-web/src/Sidebar.tsx

```diff
@@ -0,0 +1,36 @@
+import { createElement, ReactElement } from "react";
+
+import { Sidebar as SidebarComponent } from "./components/Sidebar";
+
+import { SidebarContainerProps } from "../typings/SidebarProps";
+
+export function Sidebar(props: SidebarContainerProps): ReactElement | null {
+    const width = getWidth(props.widthUnit, props.widthValue);
+    const collapsedWidth = getWidth(props.collapsedWidthUnit, props.collapsedWidthValue);
+    const expandedWidth = getWidth(props.expandedWidthUnit, props.expandedWidthValue);
+
+    return (
+        <SidebarComponent
+            className={props.class}
+            name={props.name}
+            style={props.style}
+            tabIndex={props.tabIndex}
+            collapsible={props.toggleMode !== "none"}
+            startExpanded={props.toggleMode === "none" || props.toggleMode === "startExpandedShrink"}
+            width={props.toggleMode === "none" ? width : undefined}
+            collapsedWidth={
+                props.toggleMode === "startCollapsedShrink" || props.toggleMode === "startExpandedShrink"
+                    ? collapsedWidth
+                    : undefined
+            }
+            expandedWidth={expandedWidth}
+            slideOver={props.toggleMode === "slideOver"}
```

Comment:
personally I would go with passing `toggleMode`, `collapsedWidth`, `expandedWidth` and `width` as props instead of creating boolean props for every toggle mode. Right now you're splitting one enum into 3 boolean props, which can become quite verbose and hard to read.

Just passing the enum reduces the number of props from 6 to 4, let's the `components/Sidebar` component decide what to do, and just leaves this component as a "pass through" component of values without any logic. But I can pick that up for you next sprint.

Reply:
I get what you mean. I considered both options before as well, but see now that your favoured option is the better one to go for.

Reply:
Will pick this up int he coming sprints